torch/onnx/_internal/fx/_pass.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+import abc
+import difflib
+
+from typing import Any, Callable, Dict, Generator, Sequence, Tuple
+
+import torch
+import torch.fx
+from torch.onnx._internal import _beartype
+from torch.onnx._internal.fx import diagnostics
+
+
+class FxReadableGraphDiffer(difflib.Differ):
+    """Differ specialized for fx readable graph.
+
+    The only difference between this and official `difflib.Differ` is the `autojunk`
+    argument for `difflib.SequenceMatcher` class. In this class, `autojunk` is set
+    to `False`. This is to prevent `difflib.SequenceMatcher` recognizing stacktrace
+    messages in fx readable graph as junk, as these messages tend to be long (>200)
+    and repeat multiple times, which falls under the junk filter criteria.
+
+    `Reference: Automatic junk heuristic <https://docs.python.org/3/library/difflib.html>`_
+    """
+
+    def compare(self, a: Sequence[str], b: Sequence[str]) -> Generator[str, None, None]:
+        cruncher = difflib.SequenceMatcher(None, a, b, autojunk=False)
+        for tag, alo, ahi, blo, bhi in cruncher.get_opcodes():
+            if tag == "replace":
```

Comment:
Where does those magic keywords `replace` , `delete`, `insert`, etc come from? Can you add some example links to `SequenceMatcher` so that people knows what does it really do? Just like what I did for ONNX external data.

Reply:
these are gone now